docs/src/main/asciidoc/config-extending-support.adoc

```diff
@@ -0,0 +1,172 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
+////
+= Extending Configuration Support
+
+include::./attributes.adoc[]
+
+:numbered:
+:sectnums:
+:sectnumlevels: 4
+:toc:
+
+[[custom-config-source]]
+== Custom Config Sources
+
+It is possible to create a Custom Config Sources as specified in
+link:https://github.com/eclipse/microprofile-config/blob/master/spec/src/main/asciidoc/configsources.asciidoc#custom-configsources-via-configsourceprovider[MicroProfile Config Source].
+
+With a Custom Config Source it is possible to read additional configuration values and add them to the Config
+instance in a defined ordinal. This allows overriding values from other sources or falling back to other values.
+
+image::config-sources.png[]
+
+A custom Config Source requires an implementation of `org.eclipse.microprofile.config.spi.ConfigSource` or
+`org.eclipse.microprofile.config.spi.ConfigSourceProvider`. Each implementation requires registration via
+the https://docs.oracle.com/javase/8/docs/api/java/util/ServiceLoader.html[ServiceLoader] mechanism, either in
+`META-INF/services/org.eclipse.microprofile.config.spi.ConfigSource` or
+`META-INF/services/org.eclipse.microprofile.config.spi.ConfigSourceProvider`.
+
+=== Example
+
+Consider a simple in-memory ConfigSource:
+
+.org.acme.config.InMemoryConfigSource
+[source,java]
+----
+package org.acme.config;
+
+import org.eclipse.microprofile.config.spi.ConfigSource;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+public class InMemoryConfigSource implements ConfigSource {
+    private static final Map<String, String> configuration = new HashMap<>();
+
+    static {
+        configuration.put("my.prop", "1234");
+    }
+
+    @Override
+    public int getOrdinal() {
+        return 275;
```

Comment:
Can add a comment explaining where that would be in comparison of the ENV and application.properties?

Reply:
(We may add a table somewhere actually) WDYT?

Reply:
Just saw the last sentence of the section. 
Should we add a table there?

Reply:
Sure.